server/server/src/sync_on_central/mod.rs

```diff
@@ -99,35 +96,35 @@ async fn download_file(
     Ok(response)
 }
 
-#[put("/sync/upload_file/{file_id}")]
+// Request one part 'json_part' one part 'file_part'
+// can't directly align multipart between actix_web and reqwest
+// need to be vigilant when changing parts and update equivalent upload_part in sync apiv_v6 client request
+#[derive(MultipartForm)]
+pub struct SyncUploadFileMultipartRequestV6 {
+    pub file_part: TempFile,
+    pub json_part: actix_multipart::form::json::Json<SyncUploadFileRequestV6>,
+}
+
+#[put("/sync/upload_file")]
 async fn upload_file(
-    MultipartForm(form): MultipartForm<SyncUploadFileRequestV6>,
+    MultipartForm(SyncUploadFileMultipartRequestV6 {
+        file_part,
+        json_part,
+    }): MultipartForm<SyncUploadFileMultipartRequestV6>,
     settings: Data<Settings>,
     service_provider: Data<ServiceProvider>,
-    path: web::Path<String>,
-) -> Result<HttpResponse, Error> {
-    let file_id = path.into_inner();
-    log::info!("Receiving a file via sync : {}", file_id);
+) -> actix_web::Result<impl Responder> {
+    let response = match sync_on_central::upload_file(
+        &settings,
+        &service_provider,
+        json_part.into_inner(),
+        file_part,
+    )
+    .await
+    {
+        Ok(batch) => SyncUploadFileResponseV6::Data(batch),
```

Comment:
When omSupply talk to each other, we so far used error object rather then http response codes, used the same pattern (the premise of this PR)